examples/litestar.py

```diff
@@ -196,24 +189,26 @@ async def delete_author(
         await authors_repo.session.commit()
 
 
-session_config = AsyncSessionConfig(expire_on_commit=False)
-sqlalchemy_config = SQLAlchemyAsyncConfig(
-    connection_string="sqlite+aiosqlite:///test.sqlite",
-    session_config=session_config,
-)  # Create 'db_session' dependency.
-sqlalchemy_plugin = SQLAlchemyPlugin(config=sqlalchemy_config)
-
-
-async def on_startup() -> None:
-    """Initializes the database."""
-    async with sqlalchemy_config.get_engine().begin() as conn:
-        await conn.run_sync(UUIDBase.metadata.create_all)
-
-
-app = Litestar(
-    route_handlers=[AuthorController],
-    on_startup=[on_startup],
-    plugins=[sqlalchemy_plugin],
-    dependencies={"limit_offset": Provide(provide_limit_offset_pagination, sync_to_thread=False)},
-    signature_namespace={"date": date, "datetime": datetime, "UUID": UUID},
-)
+def init_app(*, sqlalchemy_config: SQLAlchemyAsyncConfig | None = None) -> Litestar:
+    if not sqlalchemy_config:
+        # expire_on_commit=False prevents the sqlalchemy models from being invalidated on commit.
+        session_config = AsyncSessionConfig(expire_on_commit=False)
+        sqlalchemy_config = SQLAlchemyAsyncConfig(
+            connection_string="sqlite+aiosqlite:///test.sqlite",
+            session_config=session_config,
+        )  # Create 'db_session' dependency.
+
+    sqlalchemy_plugin = SQLAlchemyPlugin(config=sqlalchemy_config)
+
+    async def on_startup() -> None:
+        """Initializes the database."""
+        async with sqlalchemy_config.get_engine().begin() as conn:
+            await conn.run_sync(UUIDBase.metadata.create_all)
```

Comment:
This can be removed and the `create_all` in the config can be used.

Reply:
Fixed.

I also noticed that the monkeypatch I had in conftest.py wasn't required (just moving the import into the `app()` fixture was sufficient to prevent the problems with the other tests) so I removed conftest.py.